styx-service-common/src/main/java/com/spotify/styx/api/Authenticator.java

```diff
@@ -295,12 +297,14 @@ private boolean resolveProjectAccess(String projectId) throws IOException {
     try {
       ancestry = executeWithRetries(request, retryWaitStrategy, retryStopStrategy);
     } catch (GoogleJsonResponseException e) {
-      if (e.getStatusCode() == 404) {
+      // GCP returns 403 in case of project not found for security reason, but that makes it impossible
+      // to differentiate that from missing permission; we take the risk here assuming Styx service account does have
+      // proper permissions
+      if (e.getStatusCode() == FORBIDDEN.code()) {
         logger.debug("Project {} doesn't exist", projectId, e);
```

Comment:
```suggestion
        logger.debug("Most likely project {} doesn't exist (google will return 403 for 404 due to security), or actually have a permission issue problem", projectId, e);
```

Something better, but might be good to mention in the log too

Reply:
What I am saying here is, this feels more like a code comment than something we would want to log. I don't have strong opinion, so if we think this is better, I'm fine with that.